modules/dnn/src/layers/fast_convolution/fast_convolution.hpp

```diff
@@ -0,0 +1,47 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef OPENCV_FAST_CONVOLUTION_HPP
+#define OPENCV_FAST_CONVOLUTION_HPP
+
+namespace cv { namespace dnn {
+
+typedef struct FastConv2d
+{
+    int ngroups;
+    int K, C, Hk, Wk;
+    int stride_y, stride_x;
+    int dilation_y, dilation_x;
+    int pad_top, pad_bottom, pad_left, pad_right;
+
+    AutoBuffer<float> weightsBuf, weightsWino63Buf;
```

Comment:
in the original code it was a dedicated "free" function to deallocate FastConv2d structure and all its parts. This modification is not correct. AutoBuffer<> is intended to be used in functions. Each autobuffer takes several Kb of space in any case and is needed to minimize the memory allocation when the buffer is small most of the time. In the case of convolution the buffer for weights and especially for Winograd coefficients is usually much larger. Just use std::vector<> instead. But in this case use a regular C++ structure:

```
struct FastConv2d
{
};
```

1. do not use C-style structure: ```typedef struct FastConv2d { ... } FastConv2d;```
2. do not use extra `float *weightsPtr, *weightsWino63Ptr;` Instead, extract those pointers from vectors inside functions: ```float* weightsPtr = &weightsBuf[0];``` etc.

Reply:
fixed.